infra/modules/service/main.tf

```diff
@@ -319,3 +319,111 @@ resource "aws_security_group" "app" {
+  master_username    = "app_usr"
+  master_password    = aws_ssm_parameter.random_db_password.value
+  storage_encrypted  = true
+  # checkov:skip=CKV_AWS_128:IAM Auth will be added with the `wic-prp-eng` role created in PRP-74 https://wicmtdp.atlassian.net/browse/PRP-74
+  # checkov:skip=CKV_AWS_162:IAM Auth will be added with the `wic-prp-eng` role created in PRP-74 https://wicmtdp.atlassian.net/browse/PRP-74
+  # iam_database_authentication_enabled = true
+  deletion_protection = true
+
+
+  serverlessv2_scaling_configuration {
+    max_capacity = 1.0
+    min_capacity = 0.5
+  }
+
+}
+
+resource "aws_rds_cluster_instance" "postgresql-cluster" {
+  cluster_identifier         = aws_rds_cluster.postgresql.id
+  instance_class             = "db.serverless"
+  engine                     = aws_rds_cluster.postgresql.engine
+  engine_version             = aws_rds_cluster.postgresql.engine_version
+  auto_minor_version_upgrade = true
+  monitoring_role_arn        = aws_iam_role.rds_enhanced_monitoring.arn
+  monitoring_interval        = 30
+}
+
+resource "random_password" "random_db_password" {
+  length           = 48
+  special          = true
+  min_special      = 6
+  override_special = "!#$%&*()-_=+[]{}<>:?"
+}
+
+resource "aws_ssm_parameter" "random_db_password" {
+  name  = "/metadata/db/admin-password"
+  type  = "SecureString"
+  value = random_password.random_db_password.result
+}
+
+resource "aws_backup_plan" "postgresql" {
+  name = "${var.service_name}_backup_plan"
+
+  rule {
+    rule_name         = "${var.service_name}_backup_rule"
+    target_vault_name = "${var.service_name}-vault"
+    schedule          = "cron(0 12 ? * SUN *)"
+  }
+}
+# checkov:skip=CKV2_AWS_8:TODO add backup selection plan using tags
+
+################################################################################
+# IAM role for enhanced monitoring
+################################################################################
+
+resource "aws_iam_role" "rds_enhanced_monitoring" {
+  name_prefix        = "aurora-enhanced-monitoring-"
+  assume_role_policy = data.aws_iam_policy_document.rds_enhanced_monitoring.json
+}
+
+resource "aws_iam_role_policy_attachment" "rds_enhanced_monitoring" {
+  role       = aws_iam_role.rds_enhanced_monitoring.name
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AmazonRDSEnhancedMonitoringRole"
+}
+
+data "aws_iam_policy_document" "rds_enhanced_monitoring" {
+  statement {
+    actions = [
+      "sts:AssumeRole",
+    ]
+
+    effect = "Allow"
+
+    principals {
+      type        = "Service"
+      identifiers = ["monitoring.rds.amazonaws.com"]
+    }
+  }
+}
+
+################################################################################
+# Parameters for Query Logging
+################################################################################
+# checkov:skip=CKV2_AWS_27:have concerns about sensitive data in logs; want better way to get this information
```

Comment:
@aplybeah I agree with this comment to skip query logging for now. Can you open a ticket so we can talk through db logging as a team? Also, I'm not sure this checkov skip is working?